### PR TITLE
various optimizer fixes

### PR DESCRIPTION
**Pull request by joelberkeley**

### CODE REVIEW

File: trieste/models/model_interfaces.py [hunk 1]
```diff
@@ -63,16 +63,27 @@ def update(self, dataset: Dataset) -> None:
 
     @abstractmethod
     def optimize(self, dataset: Dataset) -> None:
-        """ Optimize the model parameters. """
+        """
+        Optimize the model parameters with respect to the specified ``dataset``.
```
Comment (vpicheny) [inline trieste/models/model_interfaces.py]: not sure this is the right wording. I mostly use "optimize objective wrt input variables".

File: trieste/models/model_interfaces.py [hunk 2]
```diff
@@ -63,16 +63,27 @@ def update(self, dataset: Dataset) -> None:
...
+
+        :param dataset: The data with which to optimize the model.
```
Comment (vpicheny) [inline trieste/models/model_interfaces.py]: train instead of optimize?

**Review by vpicheny — APPROVED** [review]

File: trieste/models/model_interfaces.py (hunk 2 above)
Comment (joelberkeley) [reply trieste/models/model_interfaces.py]: sure

File: trieste/models/model_interfaces.py (hunk 1 above)
Comment (joelberkeley) [reply trieste/models/model_interfaces.py]: good point. To include dataset parameter too, what about
```
Optimize the model objective with respect to (hyper)parameters given the specified ``dataset``.
```

**Pull request merged**